app/Http/Controllers/CourseController.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Storage;
+
+use App\Models\Course;
+use App\Repositories\CourseRepository;
+
+class CourseController extends Controller
+{
+    private $courseRepository;
+
+    public function __construct(CourseRepository $courseRepository)
+    {
+        $this->courseRepository = $courseRepository;
+    }
+
+    public function index()
+    {
+        $courses = $this->courseRepository->all();
+        return view('layout.home', compact('courses'));
+    }
+
+    public function show($id, Request $request)
+    {
+        $course = $this->courseRepository->find($id);
```

Comment:
find không có thì thế nào?

Reply:
trong CourseRepository e dùng findOrFail nên nếu ko find ko có sẽ tự return sang trang 404 Not Found

Reply:
ở đây em đang dùng find chứ finforrfail đâu nhỉ? 2 cái này khác nhau như nào?
anh muốn redirect về trang trước đó và đưa ra thông báo chứ không phải 404 thì làm như nào? xử lý lại chỗ này

Reply:
ready